pyoozie/builder.py

```diff
@@ -0,0 +1,135 @@
+        'user.name': hadoop_user,
+        'oozie.wf.application.path': hdfs_path
+    })
+    return submission.xml(indent)
+
+
+def _coordinator_submission_xml(hadoop_user, hdfs_path, configuration=None, indent=False):
+    """Generate a Coordinator XML submission message to POST to Oozie."""
+    submission = Configuration(configuration)
+    submission.update({
+        'user.name': hadoop_user,
+        'oozie.coord.application.path': hdfs_path,
+    })
+    return submission.xml(indent)
+
+
+class workflow(object):
+
+    def __init__(self, name):
+        # Initially, let's just use a static template and only one action payload and one action on error
+        self._name = name
+        self._action_name = None
+        self._action_payload = None
+        self._action_error = None
+        self._kill_message = None
+
+    def add_action(self, name, action, action_on_error, kill_on_error='${wf:lastErrorNode()} - ${wf:id()}'):
+        # Today you can't rename your action and you can only have one, but in the future you can add multiple
+        # named actions
+        if self._action_name is None and self._action_payload is None and self._action_error is None and \
+           self._kill_message is None:
+            self._action_name = name
+            self._action_payload = action
+            self._action_error = action_on_error
+            self._kill_message = kill_on_error
+        else:
+            raise NotImplementedError("Can only add one action in this version")
+        return self
+
+    def build(self, indent=False):
+        def remove_header(xml):
+            return xml.replace("<?xml version='1.0' encoding='UTF-8'?>", '')
+        return '''
+<?xml version="1.0" encoding="UTF-8"?>
+<workflow-app xmlns="uri:oozie:workflow:0.5"
+              name="{name}">
+    <start to="action-{action_name}" />
+    <action name="action-{action_name}">
+        {action_payload_xml}
+        <ok to="end" />
+        <error to="action-error" />
+    </action>
+    <action name="action-error">
+        {action_error_xml}
+        <ok to="kill" />
+        <error to="kill" />
+    </action>
+    <kill name="kill">
+        <message>{kill_message}</message>
+    </kill>
+    <end name="end" />
+</workflow-app>
+'''.format(action_payload_xml=remove_header(self._action_payload.xml(indent=indent)),
+           action_error_xml=remove_header(self._action_error.xml(indent=indent)),
+           kill_message=self._kill_message,
+           action_name=self._action_name,
+           name=self._name).strip()
+
+    def submit(self, oozie_url, hdfs_path, hadoop_user, hdfs_callback, timeout_in_seconds=None,
+               verbose=False, start=False, indent=False):
+        xml = self.build(indent=indent)
+        hdfs_callback(hdfs_path, xml)
+        # TODO create Oozie API and submit
+        from mock import Mock
+        OozieAPI = Mock()
+        api = OozieAPI(url=oozie_url, user=hadoop_user, timeout=timeout_in_seconds, verbose=verbose)
```

Comment:
Near term I think it'd make more sense to pass in a preconstructed `OozieAPI`, since the likely usecase is submitting multiple objects. Alternately, don't directly link these two at all (i.e. have the builder write to hdfs and stop).